### ISSUE TITLE
Preheat Bed and Manual + Autoleveling Bed

### ISSUE THREAD

**Issue by knopserl** [makerbase-mks/MKS-TFT#183]
How to add Bed Preheat (with temperature setting) and Autoleving additionally to Manual Bed leveling.

I need a bed preheat function in the Preheat Menu and I also wanted to have Manual + Autobed leveling in the Leveling Menu.
Is that already possible - but I just don't know how to do - or does it require enhancements?
If it is already possible by configuration, how to do it and where is a more comprehensive documentation? In the config files (+ examples) I have not found such things.

**Comment by makerbase-mks** [comment]
1、adding the bed preheat:
(1)configuration the main board firmware.
(2)configuration the TFT firmware:
#enable heated bed(yes:1; no: 0)
>cfg_custom_bed_flag:1 

2、you can selected the style of leveling:
#leveling mode(manual:0; auto:1; conceal leveling button:2)
>cfg_leveling_mode:0

**Comment by knopserl** [comment]
I have configured both as you have described but:
1.) there is no preheat bed in the TFT menu only the hot ends/extrusion 1/2
2.) I need auto bed leveling AND manual bed leveling in the menu

SO how do I configure the both options?

**Comment by makerbase-mks** [comment]
1、what is your tft?tft28 or tft32?  You should click continuously the button to button function.
2、you can configure the "More" button with command of "G29"

**Comment by knopserl** [comment]
I have a tft32 V4.0. And following the proposed settings there was no option to select preheat the bed or have manual and autobed leveing. It should be at the same display level/menu

**Comment by makerbase-mks** [comment]
Could you send me a video about this problem?

**Comment by knopserl** [comment]
I have set 2 extruders and heated bed as well as one custom "more" button and a G29 button feature.
Ther result: I can switch between Extruder E1 and Heatbed but not between E1, E2 and Heatbed.
I wanted to have Manual Leveling and Auto leveling (BOTH!) manual leveling is there, but no Autoleveling. I've trie to get it as custom buttuon but it does not show up. I have one more button but there is not action.
The Marlin firmware V2.x provided manual and autoleveling.
And Autoleveling + Manaul Leveling shall be in the Leveling menu and not somewhere else (Manual Leveling under Leveling and Autoleveling somewhere else).
And just in case ju're asking: Of course I have set two extruders (I can switch in the extrude menu between E1 and E2) but not in the preheat.

I have attached some pictues. I'm using the TFT32 abd V4.0 and firmware 3.0.2

![20180909_174237](https://user-images.githubusercontent.com/36055354/45266539-b4510680-b45c-11e8-9510-96ff359dd48e.jpg)
![20180909_174251](https://user-images.githubusercontent.com/36055354/45266540-b4510680-b45c-11e8-98cc-78469c725a5b.jpg)
![20180909_174312](https://user-images.githubusercontent.com/36055354/45266541-b4510680-b45c-11e8-9234-975a0cdc3845.jpg)
